.github/workflows/test.yml

```diff
@@ -0,0 +1,56 @@
+name: Deriv-App
+on:
+  pull_request:
+    branches:
+    - master
+env:
+  CACHE_VERSION: ${{ github.sha }}
+  COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}
+  DATADOG_CLIENT_TOKEN_LOGS: ${{ secrets.DATADOG_CLIENT_TOKEN_LOGS }}
+  DATADOG_SESSION_REPLAY_SAMPLE_RATE: ${{ secrets.DATADOG_SESSION_REPLAY_SAMPLE_RATE }}
+  DATADOG_SESSION_SAMPLE_RATE: ${{ secrets.DATADOG_SESSION_SAMPLE_RATE }}
+  DATADOG_SESSION_SAMPLE_RATE_LOGS: ${{ secrets.DATADOG_SESSION_SAMPLE_RATE_LOGS }}
+  GD_API_KEY: ${{ secrets.GD_API_KEY }}
+  GD_APP_ID: ${{ secrets.GD_APP_ID }}
+  GD_CLIENT_ID: ${{ secrets.GD_CLIENT_ID }}
+  RUDDERSTACK_PRODUCTION_KEY: ${{ secrets.RUDDERSTACK_PRODUCTION_KEY }}
+  RUDDERSTACK_STAGING_KEY: ${{ secrets.RUDDERSTACK_STAGING_KEY }}
+  RUDDERSTACK_URL: ${{ secrets.RUDDERSTACK_URL }}
+  SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+jobs:
+  build_and_test:
+    name: Build And Test
+    runs-on: Runner_16cores
```

Comment:
Running tests on `Runner_16cores` for all PRs is gonna cost a lot of money. Should we inform cost control team beforehand?

Reply:
sure, i got the verbal approval, will pass it to get the final approval to make sure about it.